### PR TITLE
Support retrying failed web requests for eventhub_client sends

### PR DESCRIPTION
**Pull request by jackgerrits**
- Retry n times before counting request as failed
- Added unit tests for retry and ensuring order
- Small refactor of authorization code in eventhub_client

### CODE REVIEW

File: reinforcement_learning/rlclientlib/logger/eventhub_client.h [hunk 1]
```diff
@@ -68,8 +102,10 @@ namespace reinforcement_learning {
     std::string _authorization;
     long long _authorization_valid_until; //in seconds
     std::mutex _mutex;
-    moving_queue<http_request_task> _tasks;
+    std::mutex _mutex_http_tasks;
```
Comment (ataymano) [inline reinforcement_learning/rlclientlib/logger/eventhub_client.h]: isn't it better to turn moving_queue into thread-safe one and have this mutex there?

Comment (ataymano) [reply reinforcement_learning/rlclientlib/logger/eventhub_client.h]: we actually had it done before (before previous moving_queue implementation turned into event_queue)

Comment (jackgerrits) [reply reinforcement_learning/rlclientlib/logger/eventhub_client.h]: The problem is that we pop and then join which may take some time. I am thinking now that the synchronization may need to occur at a slightly greater scope. Let me chat with you offline.

**Comment by rajan-chari** [comment]
Merged Thanks

**Pull request merged**